client/src/main/java/com/linecorp/decaton/client/internal/DecatonTaskProducer.java

```diff
@@ -54,20 +52,13 @@ private static Properties completeProducerConfig(Properties producerConfig) {
         return result;
     }
 
-    public DecatonTaskProducer(String topic, Properties producerConfig,
+    public DecatonTaskProducer(Properties producerConfig,
                                KafkaProducerSupplier producerSupplier) {
         Properties completeProducerConfig = completeProducerConfig(producerConfig);
         producer = producerSupplier.getProducer(completeProducerConfig);
-        this.topic = topic;
     }
 
-    public CompletableFuture<PutTaskResult> sendRequest(byte[] key, DecatonTaskRequest request,
```

Comment:
So is this class now essentially a slightly different KafkaProducer implementation which at least gives CF as the returning value of sendRequest? Does it worth it to keep the class itself then..?

Reply:
It also fill preset configs (e.g. acks=all).
So I think it's worth to keep it

Reply:
okay